### PR TITLE
treat <script type="text/x-template"> as html

### LINKED ISSUE #1200: 🐛Assets in HTML template in script with type "x-template" won't get bundled
### 🎛 Configuration (.babelrc, package.json, cli command)

No configuration, I just `parcel index.html`

### 🤔 Expected Behavior
I think assets in "x-template" script tag should also be bundled. 
Code:
```html
  <script type="text/x-template" id="template">
<div>
  <img src="./static/img.png" />
</div>
</script>
```
### 😯 Current Behavior

`img.png` won't get bundled. 

### 💁 Possible Solution

Don't konw

### 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           |  v1.7.1
| Node             | v9.7.1
| npm/Yarn         | yarn
| Operating System | macOS High Sierra 10.13.3

### PR DESCRIPTION
**Pull request by makmm**
fixes #1200

---
name: 🙋 New Pull Request
about: Do you want to change something on parcel?
---

## ↪️ Pull Request

fixes #1200

## 💻 Examples

```
<script type="text/x-template" id="template">
  <div>
    <img src="./static/img.png" />
  </div>
</script>
```
will properly import `./static/img.png`

### CODE REVIEW

File: src/assets/HTMLAsset.js [hunk 1]
```diff
@@ -67,7 +67,8 @@ const SCRIPT_TYPES = {
   'text/javascript': 'js',
   'application/json': false,
   'application/ld+json': 'jsonld',
-  'text/html': false
+  'text/html': false,
+  'text/x-template': 'html'
```
Comment (DeMoorJasper) [inline src/assets/HTMLAsset.js]: Shouldn't this be just false if it's html? It's probably the same as `text/html` right?

Comment (DeMoorJasper) [reply src/assets/HTMLAsset.js]: Should have looked at the issue first... Thanks for fixing this.

**Comment by devongovett** [comment]
Is `text/x-template` an actual standard? Not sure if we should be inventing our own script types.

**Comment by makmm** [comment]
@devongovett I don't really know. I just tried solving the issue, but didn't check for standards. Do you want me to properly add tests, even if it isn't standard?

**Comment by Hoishin** [comment]
`x-template` isn't standard. I think it should be treated as plain text (as it says, **text**/x-template), rather than treating it as HTML.

Most of the time when you want to treat `x-template` as HTML, you are doing something wrong.

**Comment by makmm** [comment]
@Hoishin The thing is, the x-template thing is used for stuff like Handlebars, and inside that template you DO want to bundle assets.

**Comment by Hoishin** [comment]
`text/(x-)template` is expected to be read by template engine you use, and it really depends on the template engine. It might be HTML, Pug-like syntax, mastache syntax, or anything the template engine expects.
If there was standard, bundlers could process it as per the spec, but just treating it as HTML would introduce unexpected behaviour.

**Comment by devongovett** [comment]
Closing. I don't think there is a good way to support this in a generic way.

**Pull request closed without merging**